### DISCUSSION TITLE
ChildStack in state example

### DISCUSSION THREAD

**Discussion by Knoxvillekm** [arkivanov/Decompose#274] [Q&A]
I am using kotlin mvi and want to use childStack in store's state.

```
data class State(
     val param1: Any,
     val param2: Any,
     val childStack: ChildStack<Configuration, Child> // without value
)
```

**Comment by arkivanov** [comment]
Hello. Since you posted in Q&A, do you want to ask a specific question?

**Comment by Knoxvillekm** [comment]
How to use ChildStack without Value to change state by Reducer?

**Reply by arkivanov** [reply]
Got it thanks! It looks like you are trying to have a single state for your app, where all screens and their own states are part of one big state. Decompose encourages another approach - to split (decompose) the project into independent, reusable and lifecycle-aware blocks (components), and to combine them as needed. You can scope a dedicated MVI state (MVI store) inside each component.

**Comment by Knoxvillekm** [comment]
How to use logging functionality and time travel feature for ChildStack?

**Reply by arkivanov** [reply]
If you mean MVIKotlin, then the time travelling feature is not supposed to affect navigation. That's kinda a tradeoff in favour of reusable lifecycle-aware features. You can navigate manually to a required screen, then record time travel events there. After that you can replay and debug that screen.

Same for logging, MVIKotlin won't log your navigation. But you can log it manually when you navigate with Decompose.

**Reply by Knoxvillekm** [reply]
Thanks for the explanation. Decided to refuse to use MVIKotlin.
